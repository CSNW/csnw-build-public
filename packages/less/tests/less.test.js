const { join } = require('path');
const { Files } = require('@csnw-build/pipeline');

const less = require('../')();
const cwd = join(__dirname, 'fixtures');

test('pipeline', async () => {
  const process = await less.pipeline({
    cwd,
    sourcemaps: true,
    cache: false
  });
  const files = await Files.from('*.less', { cwd });

  const result = await process(files);
  expect(result).toMatchSnapshot();
});
