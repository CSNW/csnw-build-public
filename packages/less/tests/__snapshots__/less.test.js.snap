// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pipeline 1`] = `
Object {
  "a.css": Object {
    "baseDir": "C:/dev/csnw-build/packages/less/tests/fixtures",
    "contents": "body {
  background-color: blue;
}
h1 {
  color: red;
}
",
    "cwd": "C:/dev/csnw-build/packages/less/tests/fixtures",
    "path": "C:/dev/csnw-build/packages/less/tests/fixtures/a.css",
    "sourceMap": Object {
      "file": "a.css",
      "mappings": "AAEA;EACE,sBAAA;;ACHF;EACE,UAAA",
      "names": Array [],
      "sources": Array [
        "a.less",
        "_b.less",
      ],
      "sourcesContent": Array [
        "@color-bg: blue;

body {
  background-color: @color-bg;
}

@import '_b';
",
        "h1 {
  color: red;
}
",
      ],
      "version": 3,
    },
  },
}
`;
